Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm

```diff
@@ -262,24 +262,35 @@ static String defaultWebsiteDataStoreRootDirectory()
 
 void WebsiteDataStore::fetchAllDataStoreIdentifiers(CompletionHandler<void(Vector<UUID>&&)>&& completionHandler)
 {
-    Vector<UUID> identifiers;
-    for (auto identifierString : FileSystem::listDirectory(defaultWebsiteDataStoreRootDirectory())) {
-        if (auto identifier = UUID::parse(identifierString))
-            identifiers.append(*identifier);
-    }
-
-    completionHandler(WTFMove(identifiers));
+    websiteDataStoreIOQueue().dispatch([completionHandler = WTFMove(completionHandler), directory = defaultWebsiteDataStoreRootDirectory().isolatedCopy()]() mutable {
+        Vector<String> identifierStrings;
+        for (auto identifierString : FileSystem::listDirectory(directory))
+            identifierStrings.append(identifierString);
+        RunLoop::main().dispatch([completionHandler = WTFMove(completionHandler), identifierStrings = crossThreadCopy(WTFMove(identifierStrings))]() mutable {
+            Vector<UUID> identifiers;
+            for (auto& identifierString : identifierStrings) {
+                if (auto identifier = UUID::parse(identifierString))
```

Comment:
We should do this on the background thread I think. It would have 2 benefits:
1. We do more work on the background thread and less on the main thread
2. We wouldn't need to isolate copy the vector since UUIDs are 128-bit integers.